backend/hitas/views/housing_company.py

```diff
@@ -440,3 +442,49 @@ def get_detail_queryset(self):
     @staticmethod
     def get_filterset_class():
         return HousingCompanyFilterSet
+
+    @action(detail=True, methods=["PATCH"], url_path="batch-complete-apartments")
+    def batch_complete_apartments(self, request, **kwargs) -> Response:
+        housing_company_uuid = lookup_id_to_uuid(self.kwargs["uuid"], HousingCompany)
+
+        input_data = BatchCompleteApartmentsSerializer(data=request.data)
+        input_data.is_valid(raise_exception=True)
+
+        data = input_data.validated_data
+
+        query_set = Apartment.objects.filter(
+            building__real_estate__housing_company__uuid=housing_company_uuid,
+            completion_date=None,
```

Comment:
It's intentional. The scenario is that you wouldn't want to change a completed housing company's completion date accidentally. E.g. when apartments 1-20 are completed first and then 21-40, if you accidentally put 20-40 the second time, apartment number 20 will be changed to the new completion date.

I guess it would be ok to allow changing the date even after it's set since it's just another API call from this endpoint to fix those mistakes. Also, it will allow the added behavior of clearing the completion date. I'll change this.

Reply:
Done